lib/utils.php

```diff
@@ -1,19 +1,29 @@
 <?php
-//	Interrogates $_GET
+
+// convert all non printable ascii character to a dot '.'
+function ascii_only( $str )
+{
+    // \x20 = space; \x7e = '~' 
+    return preg_replace('/[^\x20-\x7e]/', '.', $str);
+}
```

Comment:
I think what you really want here is utf8 sanitization. Instead of blindly discarding all non-ASCII characters, you should only convert characters that don't conform to the UTF-8 multibyte pattern. Consider using `mb_convert_encoding`:
https://www.sitepoint.com/community/t/how-to-sanitize-utf-8-input-efficiently/4801